packages/surveys/src/components/general/element-conditional.tsx

```diff
@@ -0,0 +1,293 @@
+        <PictureSelectionQuestion
+          key={element.id}
+          question={element}
+          value={Array.isArray(value) ? value : []}
+          onChange={onChange}
+          languageCode={languageCode}
+          ttc={ttc}
+          setTtc={setTtc}
+          autoFocusEnabled={autoFocusEnabled}
+          currentQuestionId={currentElementId}
+          dir={dir}
+        />
+      ) : element.type === TSurveyElementTypeEnum.FileUpload ? (
+        <FileUploadQuestion
+          key={element.id}
+          surveyId={surveyId}
+          question={element}
+          value={Array.isArray(value) ? value : []}
+          onChange={onChange}
+          onFileUpload={onFileUpload}
+          languageCode={languageCode}
+          ttc={ttc}
+          setTtc={setTtc}
+          autoFocusEnabled={autoFocusEnabled}
+          currentQuestionId={currentElementId}
+        />
+      ) : element.type === TSurveyElementTypeEnum.Cal ? (
+        <CalQuestion
+          key={element.id}
+          question={element}
+          value={typeof value === "string" ? value : ""}
+          onChange={onChange}
+          languageCode={languageCode}
+          ttc={ttc}
+          setTtc={setTtc}
+          currentQuestionId={currentElementId}
+        />
+      ) : element.type === TSurveyElementTypeEnum.Matrix ? (
+        <MatrixQuestion
+          question={element}
+          value={typeof value === "object" && !Array.isArray(value) ? value : {}}
+          onChange={onChange}
+          languageCode={languageCode}
+          ttc={ttc}
+          setTtc={setTtc}
+          currentQuestionId={currentElementId}
+        />
+      ) : element.type === TSurveyElementTypeEnum.Address ? (
+        <AddressQuestion
+          question={element}
+          value={Array.isArray(value) ? value : undefined}
+          onChange={onChange}
+          languageCode={languageCode}
+          ttc={ttc}
+          setTtc={setTtc}
+          currentQuestionId={currentElementId}
+          autoFocusEnabled={autoFocusEnabled}
+          dir={dir}
+        />
+      ) : element.type === TSurveyElementTypeEnum.Ranking ? (
+        <RankingQuestion
+          question={element}
+          value={Array.isArray(value) ? getResponseValueForRankingQuestion(value, element.choices) : []}
+          onChange={onChange}
+          languageCode={languageCode}
+          ttc={ttc}
+          setTtc={setTtc}
+          autoFocusEnabled={autoFocusEnabled}
+          currentQuestionId={currentElementId}
+        />
+      ) : element.type === TSurveyElementTypeEnum.ContactInfo ? (
+        <ContactInfoQuestion
+          question={element}
+          value={Array.isArray(value) ? value : undefined}
+          onChange={onChange}
+          languageCode={languageCode}
+          ttc={ttc}
+          setTtc={setTtc}
+          currentQuestionId={currentElementId}
+          autoFocusEnabled={autoFocusEnabled}
+          dir={dir}
+        />
+      ) : null}
```

Comment:
we just render nothing when the element.type is not listed here. Should we at least log a warning?